File: Solutions/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character,Integer> m=new HashMap<>();
        int res=0;
        for(int i=0,j=0;j<s.length();++j){
            if(m.containsKey(s.charAt(j)))
                i=Math.max(i,m.get(s.charAt(j))+1);
            m.put(s.charAt(j),j);
            res=Math.max(res,j-i+1);
        }
        return res;
    }
}

File: Solutions/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.go
func lengthOfLongestSubstring(s string) int {
    m:=make(map[string]int)
    res:=0
    for i,j:=0,0;j<len(s);j+=1{
        if pos,ok:=m[string(s[j])];ok{
            i=max(i,pos+1)
        }
        m[string(s[j])]=j
        res=max(res,j-i+1)
    }
    return res
}
func max(i,j int) int{
    if i>j{
        return i
    }
    return j
}